### DISCUSSION TITLE
How do we load sample/specimen data using bulk upload

### DISCUSSION THREAD

**Discussion by Mullai** [broadinstitute/seqr#2310] [Q&A]
Hello,

We are pilot testing a local install of seqr and were successfully able to bulk upload families and individuals but are unable to find a similar option for uploading specimen data. Appreciate input in this regard - Is this something we missed in our install or is this functionality located elsewhere?

Thanks,
Mullai

![Screen Shot 2021-11-18 at 7 45 48 PM](https://user-images.githubusercontent.com/7500861/142550825-4598c292-7a2f-4784-b989-db2f13d89847.png)
a

**Comment by hanars** [comment]
I'm not 100% sure what you mean by "specimen data" but probably its in "Bulk Edit Individuals"

**Reply by Mullai** [reply]
Apologies for not being clear. Bulk Edit Individuals only includes information about the individual i.e. family id, individual id, paternal id, maternal id, sex, affected status and notes. We are looking for an option where we can upload sample information for an individual i.e. family id, individual id, sample id, sample collected date, sample type etc.

**Reply by hanars** [reply]
This is not data seqr currently stores in a structured way. When you actually load data, where the sample id should match the individual id, it will add basic metadata to seqr about whether a sample has WES or WGS data, but beyond that the data you are asking to store can only be stored as free text notes

**Reply by Mullai** [reply]
I am a little thrown off since I saw this template (attached) while testing bulk loading data seqr but was never able to find this option again. To confirm - seqr assumes individual equals sample and there is not an option to load two samples for one individual with the possible intention of using one sample for WGS and another for WES? 
[mmtest_samples.csv](https://github.com/broadinstitute/seqr/files/7589256/mmtest_samples.csv)

**Reply by hanars** [reply]
I'm not really sure where you found that template, but that is not how sample data is loaded to seqr. Data is loaded to seqr by running the loading pipeline on a joint called VCF, and then adding the elasticsearch index generated by that pipeline to the desired seqr project. By adding the index to a project, records for all the samples in that index will be generated automatically. When you go to add the index, there is the option to add an optional mapping file which will allow you to map the sample IDs in the VCF/ index to the individual IDs in seqr, if those differ. However, we advise strongly against using this functionality, as there is really no reason not to have the individual IDs align with the sample IDs. You can always change the individual Is by using the "Bulk edit individuals" options

It is theoretically possible to load WES and WGS data for the same "sample", but it is worth noting that we do not do that in our instance of seqr so I can't guarantee that that will 100% behave the way you want it to. We maintain separate projects for WES and WGS data. That said, if you add an elasticsearch index generated from WES data to a project, and then add an index generated from WGS data to the same project, it will allow you to have 1 individual associated with the 2 samples (seqr will consider them to be 2 different samples even if they were collected form the same underlying biological sample)

**Reply by Mullai** [reply]
Thank you for the detailed explanation, your response answers my question and also provides a clearer view into seqr's functionality.
